inc/packages/namespace.php

```diff
@@ -0,0 +1,592 @@
+			);
+		}
+	}
+
+	// Set a flag for whether this is site level or network level.
+	$for_network = is_network_admin();
+
+	// Track whether the index needs to be updated after any package updates.
+	$should_update_indexes = false;
+
+	// The different types of package.
+	$types = [ 'synonyms', 'stopwords', 'user-dictionary' ];
+
+	foreach ( $types as $type ) {
+		$text_field = "{$type}-text";
+		$file_field = "{$type}-file";
+		$remove_field = "{$type}-remove";
+		$package_id = null;
+
+		// Handle manual entry.
+		if ( ! empty( $_POST[ $text_field ] ) ) {
+			$text = sanitize_textarea_field( wp_unslash( $_POST[ $text_field ] ) );
+			$file = get_package_path( "manual-{$type}", $for_network );
+			$has_changed = true;
+
+			// Check for updates.
+			if ( file_exists( $file ) ) {
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+				$has_changed = $text !== file_get_contents( $file );
+			}
+
+			// Write to uploads.
+			if ( $has_changed ) {
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
+				$result = file_put_contents( $file, $text );
+				if ( ! $result ) {
+					// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+					$errors[] = new WP_Error( 'write_package_error', sprintf( __( 'Could not write search package file to %s', 'altis' ), $file ) );
+				} else {
+					$package_id = create_package( "manual-{$type}", $file, $for_network );
+					if ( is_wp_error( $package_id ) ) {
+						$errors[] = $package_id;
+					}
+				}
+			}
+		}
+
+		// Handle file upload.
+		if ( ! empty( $_FILES ) && isset( $_FILES[ $file_field ] ) && ! empty( $_FILES[ $file_field ]['tmp_name'] ) ) {
+			$file = get_package_path( "uploaded-{$type}", $for_network );
+			$result = move_uploaded_file( $_FILES[ $file_field ]['tmp_name'], $file );
+			if ( ! $result ) {
+				// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+				$errors[] = new WP_Error( 'write_package_error', sprintf( __( 'Could not write search package file to %s', 'altis' ), $file ) );
+			} else {
+				$package_id = create_package( "uploaded-{$type}", $file, $for_network );
+				if ( is_wp_error( $package_id ) ) {
+					$errors[] = $package_id;
+				} else {
+					// User dictionary update means we should update the indexed data.
+					if ( $type === 'user-dictionary' ) {
+						$should_update_indexes = true;
+					}
+				}
+			}
+		}
+
+		// Delete file if remove submit clicked.
+		if ( isset( $_POST[ $remove_field ] ) ) {
+			$deleted = delete_package( "uploaded-{$type}", $for_network );
+			if ( is_wp_error( $deleted ) ) {
+				$errors[] = $deleted;
+			} else {
+				// User dictionary update means we should update the indexed data.
+				if ( $type === 'user-dictionary' ) {
+					$should_update_indexes = true;
+				}
+			}
+		}
+	}
+
+	// Store the errors.
+	array_map( __NAMESPACE__ . '\\add_error_message', $errors );
```

Comment:
Might be clearer that we're not expecting to use the return value, plus benefits from type hinting at compile time:

```suggestion
	foreach ( $errors as $error ) {
		add_error_message( $error );
	}
```